database/factories/Company/CompanyInvoiceFactory.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace Database\Factories\Company;
+
+use App\Models\Company\Company;
+use App\Models\Company\CompanyInvoice;
+use App\Models\Company\CompanyDailyUsageHistory;
+use Illuminate\Database\Eloquent\Factories\Factory;
+
+class CompanyInvoiceFactory extends Factory
+{
+    /**
+     * The name of the factory's corresponding model.
+     *
+     * @var string
+     */
+    protected $model = CompanyInvoice::class;
+
+    /**
+     * Define the model's default state.
+     *
+     * @return array
+     */
+    public function definition()
+    {
+        return [
+            'company_id' => Company::factory(),
+            'usage_history_id' => CompanyDailyUsageHistory::factory(),
```

Comment:
Shouldn't it be something like
```suggestion
            'usage_history_id' => function (array $attributes) {
                return CompanyDailyUsageHistory::factory()->create([
                    'company_id' => $attributes['company_id'],
                ]);
            }),
```

Reply:
@asbiin definitely yes! If that works! I wondered how to achieve this.